iwyu_preprocessor.cc

```diff
@@ -751,6 +758,23 @@ void IwyuPreprocessorInfo::FileChanged_ExitToFile(SourceLocation include_loc,
          "begin_exports without an end_exports");
     begin_exports_location_stack_.pop();
   }
+
+  const FileEntry* includer = GetFileEntry(associated_pragma_location_);
+  if (associated_pragma_location_.isValid() &&
+	  includer == GetFileEntry(include_loc)) {
+	  GetFromFileInfoMap(includer)
+		  ->AddAssociatedHeader(GetFromFileInfoMap(exiting_from));
+	  VERRS(4) << "Marked " << GetFilePath(exiting_from)
+		  << " as associated header of " << GetFilePath(includer) << ".\n";
+
+	  AddGlobToReportIWYUViolationsFor(GetFilePath(exiting_from));
+	  if (ShouldReportIWYUViolationsFor(exiting_from)) {
```

Comment:
Yes it is unnecessary, i took this line from the code i used as example. At that place it was correct, here it is unnecessary. Will change in the next commit.